client/v1/impl.go

```diff
@@ -39,15 +39,23 @@ func newFullHTTPClient(cfg *ClientConfig) HTTPRootClient {
 		}
 	}
 
+	headers := make(http.Header)
+
+	if cfg.authToken != "" {
+		headers.Set("Authorization", fmt.Sprintf("Bearer %s", cfg.authToken))
+	}
```

Comment:
I'd say that instead of handling authentication header manually, we should instead rely on the ability to pass an `*http.Client` to this. e.g. using this wrapper: https://pkg.go.dev/golang.org/x/oauth2#NewClient

Reply:
I'm generally in favor of that idea. One caveat is that we'll want to implement our own `TokenSource` implementation, but I don't think that's too hard.

Reply:
Makes sense. This has now been removed and oauth2.NewClient should be passed using `WithClient()`